### PR TITLE
Fix doxygen documentation for release

### LINKED ISSUE #865: Documentation for release 0.1
Review and publish the documentation for release 0.1

Part of #630

### PR DESCRIPTION
**Pull request by rasolca**
All the files with public API are now parsed correctly and fixed a couple of formatting problems.
Closes #865

### CODE REVIEW

**Comment by rasolca** [comment]
cscs-ci run

File: include/dlaf/blas/tile.h [hunk 1]
```diff
@@ -162,23 +165,21 @@ using matrix::Tile;
...
 template <Backend B, class T, Device D>
 void gemm(const blas::Op op_a, const blas::Op op_b, const T alpha, const Tile<const T, D>& a,
           const Tile<const T, D>& b, const T beta, const Tile<T, D>& c);
 
-/// \overload gemm
-///
 /// This overload takes a policy argument and a sender which must send all required arguments for the
 /// algorithm. Returns a sender which signals a connected receiver when the algorithm is done.
+///
+/// Test
```
Comment (msimberg) [inline include/dlaf/blas/tile.h:178]:
```suggestion
```

File: include/dlaf/blas/tile_extensions.h [hunk 2]
```diff
@@ -39,16 +42,12 @@ using matrix::Tile;
 template <Backend B, class T, Device D>
```
Comment (msimberg) [inline include/dlaf/blas/tile_extensions.h:42]: Do we need `@brief` on the first line of the docstring above? Edit: actually, is it needed in general? I just saw it on the tile blas overloads which is what made me comment, but it's not used on most functions/classes. Is it only required to have the `@brief` description show up on the other overloads?

**Review by msimberg — CHANGES_REQUESTED** [review]

File: include/dlaf/blas/tile_extensions.h (hunk 2 above)
Comment (rasolca) [reply include/dlaf/blas/tile_extensions.h:42]: We use the syntax where `@brief` is not needed:
```
/// Brief doc
///
/// Full doc
```

**Review by msimberg — APPROVED** [review]

File: include/dlaf/common/assert.h
Comment (albestro) [inline include/dlaf/common/assert.h:1]: Just as placeholder for future reference. AFAIU doxygen is not able to generate the correct documentation for this file.

**Review by albestro — APPROVED** [review]

**Pull request merged**